### ISSUE TITLE
Speed

### ISSUE THREAD

**Issue by sriramkukillaya** [joncampbell123/dosbox-x#3595]
### Is your feature request related to a problem? Please describe.

Speed of dosbox x needs to go smooth like dosbox 0.70

### What you want

Speed lags in dosbox x . Please mske it go smooth like dosbox 0.63 and 0.70

### Have you checked that no similar feature request(s) exist?

- [X] I have searched and didn't find any similar feature request.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by sriramkukillaya** [comment]
Speed needs to be like dosbox 0.63 when I play wolfenstein3d and like 0.70 wjen I play Doom or Duke3d

**Comment by rderooy** [comment]
Improved emulation is not free.

**Comment by rderooy** [comment]
Not free, as in requiring more CPU overhead.

**Comment by sriramkukillaya** [comment]
Ok, thanks for the reply.  Cause when I put dosbox 0.63 to 20000 on my old
pc , it ran smoothly but when I increase it on dosbox x it lags . If you
can improve these elements on the next project of yours please .

**Comment by sriramkukillaya** [comment]
On the new pc also

**Comment by sriramkukillaya** [comment]
Please make these improvements in the new version

**Comment by sriramkukillaya** [comment]
One thing in dosbox 0.063 to 0.74.3 is that it doesn't get stretched

**Comment by Jookia** [comment]
Did you try setting cycles=max

**Comment by sriramkukillaya** [comment]
Yes, I did. But didn't work . Please add irq / interrupt 2 , 3 , 5 , 7 .

**Comment by Jookia** [comment]
For what? Soundblaster?

**Comment by sriramkukillaya** [comment]
Sorry I set it to 20000 in dosbox 0.63

**Comment by sriramkukillaya** [comment]
Yes sound blaster . I set the CPU cycles in Dosbox 0.63 .

**Comment by sriramkukillaya** [comment]
Yes for the Sound Blaster please . I also set the CPU cycles in Dosbox 0.63
to 20000 . So that's how it worked

**Comment by sriramkukillaya** [comment]
There was no max in 0.63 Dosbox

**Comment by Jookia** [comment]
You can set the sound blaster IRQs to one of these values: 7, 5, 3, 9, 10, 11, 12, 0, -1

**Comment by sriramkukillaya** [comment]
I thought you can do these for the next project .

**Comment by Jookia** [comment]
If there's a use case for it maybe you could explain here?

**Comment by sriramkukillaya** [comment]
I just miss it . When I put it to max for funschool 4 and when I go the
main menu after playing a game , the sound and speed lag . Improvements
needed please . Let it go to max for doom , duke3d etc .

**Comment by Jookia** [comment]
Sorry I don't understand what you're saying.

**Comment by sriramkukillaya** [comment]
No problem . What I meant was that when I set the interrupt to value 2 on the Audio Software for dos and when I loaded wolf3d and when I go near any boss ,  the gun sound would get cut and I liked that . I was wondering whether you can include these changes .

**Comment by rderooy** [comment]
IRQ2 was never supported by SoundBlasters.

On a AT or newer machine type, that is the cascade IRQ for the second interrupt controller and should not be used.

**Comment by sriramkukillaya** [comment]
Okay cause I remember my old machine in 1987 where I had the interrupt at 2
and when I downloaded PC Emulator , I installed the Audio Software and I
set the interrupt to 2 for my wolf3d

**Comment by rderooy** [comment]
Ok, you are indeed right, on the old 8bit SoundBlasters, IRQ2 was an option. But this was only for PC-XT class machines with a single PIC (Programmable Interrupt Controller). And even then, IRQ2 is often used for Video on those systems.

So, if you select an 8-bit SB (e.g. SB 1.5) and have a XT machine type with a single PIC, it should be a valid option.

**Comment by sriramkukillaya** [comment]
Thank you so much . Is it available on doxbox x

**Comment by rderooy** [comment]
The option is not available. The options are:
```
irq: The IRQ number of the Sound Blaster (usually 5 or 7, depending on the sound card type and the game).
#                                                     Set to 0 for the default setting of the sound card, or set to -1 to start DOSBox-X with the IRQ unassigned.
#                                                     Possible values: 7, 5, 3, 9, 10, 11, 12, 0, -1.
```
I think the reason that IRQ2 is not offered, is inherited from vanilla DOSBox which does not support PC-XT machine types with a single PIC. So offering IRQ2 makes little sense for them.

I honestly don't think there is any benefit to offering IRQ2. And if it were to be offered, it would have to be only for select cases (SB models that actually supported it).

Whatever your problem is, is likely to be some other issue. Wolfenstein 3D works just fine here with IRQ5 or IRQ7.

**Comment by rderooy** [comment]
According to this, IRQ2 was offered on all the SoundBlaster cards up to and including the original SB16
http://nerdlypleasures.blogspot.com/2010/06/issues-with-sound-blasters.html

Still I think it is not offered in DOSBox (and therefore DOSBox-X), is because of the conflict it causes on PC-AT machines. They probably figured that offering IRQ2 would just increase support issues.

**Comment by sriramkukillaya** [comment]
Okay

Any other solution ?

**Comment by sriramkukillaya** [comment]
I forgot to mention one thing . In Dosbox 0.63 there's no automatic max
available . It had to be set manually and I increased it to 20000 for one
version of my wolfenstein 3d game so it went pretty fast . Please try and
make these changes if possible .

Thanks

**Comment by sriramkukillaya** [comment]
I mean auto cpu cycles for Doom Duke3d , Wacky , Jazz Jackrabbit etc

**Comment by kcgen** [comment]
Hi Sriram,
What operating system, version, and hardware platform (CPU and GPU specs) are you running these games on?

**Comment by sriramkukillaya** [comment]
Greetings
I am using a lenovo ideapad 110 and AMD A6 . I put an 8gb ram .

**Comment by sriramkukillaya** [comment]
Windows 10

**Comment by kcgen** [comment]
Thanks for those details, @sriramkukillaya.

Maybe you used a different machine with dosbox 0.63? 

The A6 CPU in that laptop is a lower power model roughly on part with Celeron and atom-based Chromebooks, meaning it will struggle to run the heavier late-90s 3D games at high resolutions and refresh rates. 

It's roughly as fast as a raspberry Pi 4 in some tests, and AMD support for 3D drivers has expired years ago (unfortunately): 
https://www.amd.com/en/support/apu/amd-series-processors/amd-a6-series-apu-for-laptops/a6-7310-radeon-r4-graphics

That said, DOSBox-X release binaries are built with optimizations and your CPU should still get you between 30k and 100k cycles. 

Can you try running the benchmarks in this package: https://anttipeltola.eu/misc/files/DOSBENCH.ZIP , and report the results back here with screenshots?

**Comment by sriramkukillaya** [comment]
Thank you so much

In 0.63 it doesn't stretch into full screen .

In X it gets stretched . I was wondering whether you may be able to do a
combination of 0.63 . 0.65 and 0.74 for your next project .

**Comment by sriramkukillaya** [comment]
One more thing
 Alt + Enter for fullcreen and Ctrl + F11 to reduce the cpu cycles and F12
to Increase it .

**Comment by kcgen** [comment]
> In 0.63 it doesn't stretch into full screen .

Use Fit-to-Aspect and a 3:4 ratio:

![Screenshot from 2022-08-01 07-17-12](https://user-images.githubusercontent.com/1557255/182168297-bf517244-341d-433a-9ca6-9d0018e8302b.png)

**Comment by sriramkukillaya** [comment]
If you can make the speed of dosbox x like 0.63 , 0.65 and 0.74 .
Combination for games please .

**Comment by sriramkukillaya** [comment]
Leave the sound part alone please at least can you do something about the
Speed but keep the max setting as well

**Comment by sriramkukillaya** [comment]
Thanks, I'll check it out.

Please improve the speed alone and in my old hp computer i used 0.63 and it
was the same like in the video which I sent of my self playing wolf3d

**Comment by symbios24** [comment]
Hello there I have a raspberry pi4 and installed windows 98 the performance is very very slow

As i read older versions 0.63 and 0.70 are running smooth how can i compile 0.63 or 0.70 versions as i cannot seem to find the source code in releases

Thank you

**Comment by rderooy** [comment]
@symbios24 Please don't cross-post your questions into multiple issues.

**Comment by symbios24** [comment]
Ok sorry 😐
